test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -3618,6 +3620,160 @@ public async Task Restore_WithTransitiveProjectReference_WhenTransitiveProjectRe
             result.LockFile.LogMessages.Single(e => e.LibraryId == "a");
         }
 
+        [Fact]
+        public async Task Restore_WhenSourceDoesNotExist_ReportsNU1301InAssetsFile()
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+
+            // Don't create the package
+            var packageA = new SimpleTestPackageContext("a", "1.0.0");
+            var logger = new TestLogger();
+            Directory.Delete(pathContext.PackageSource);
+            // Set-up command.
+            var command = new RestoreCommand(
+                ProjectTestHelpers.CreateRestoreRequest(
+                    ProjectTestHelpers.GetPackageSpec("Project1", pathContext.SolutionRoot, framework: "net5.0", dependencyName: packageA.Id),
+                    pathContext,
+                    new TestLogger()));
+
+            // Act
+            var result = await command.ExecuteAsync();
+
+            // Assert
+            result.Success.Should().BeFalse(because: logger.ShowMessages());
+            result.LockFile.Libraries.Should().HaveCount(0);
+            result.LockFile.LogMessages.Should().HaveCount(1);
+            result.LockFile.LogMessages.Select(e => e.Code).Should().AllBeEquivalentTo(NuGetLogCode.NU1301);
+        }
+
+        [Fact]
+        public async Task Restore_WhenSourceDoesNotExist_AndIgnoreFailedSourcesIsTrue_ReportsNU1801InAssetsFile()
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            var extraSource = Path.Combine(pathContext.WorkingDirectory, "Source2");
+            pathContext.Settings.AddSource("extraSource", extraSource);
+            var packageA = new SimpleTestPackageContext("a", "1.0.0");
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(
```

Comment:
Is `IgnoreFailedSources`  set to true here?

Reply:
Yes, CreateRestoreRequest does it.